src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -456,6 +474,29 @@ void ProcessMarkedTypesWithInterfaces ()
 			}
 		}
 
+		void ProcessDynamicCastableImplementationInterfaces ()
+		{
+			// We may mark an interface type later on.  Which means we need to reprocess any time with one or more interface implementations that have not been marked
+			// and if an interface type is found to be marked and implementation is not marked, then we need to mark that implementation
+
+			for (int i = 0; i < _dynamicInterfaceCastableImplementationTypes.Count; i++) {
+				var type = _dynamicInterfaceCastableImplementationTypes[i];
+
+				Debug.Assert (TypeHasDynamicInterfaceCastableImplementationAttribute (type));
+
+				if (Annotations.IsMarked (type))
+					continue;
+
+				foreach (var iface in type.Interfaces) {
+					if (Annotations.IsMarked (iface.InterfaceType)) {
+						MarkType (type, new DependencyInfo (DependencyKind.DynamicInterfaceCastableImplementation, iface.InterfaceType), type);
+						_dynamicInterfaceCastableImplementationTypes.RemoveAt (i--);
```

Comment:
Do we really want to play the post-decrement side effect game? I don't think this costs much to pass in `i` and then `i--` in the next statement with a comment. Alternative reworking this logic to avoid manipulating a collection under inspection seems like the right thing to do _or_ even collecting the indices to remove and removing them post loop would be preferable in my opinion.

Reply:
Oh... This is a common pattern in here. Hmmm I guess never mind.

Reply:
Yeah it's not my favorite pattern, but it works and the rest of the code base uses it.